yt_dlp/extractor/hidive.py

```diff
@@ -57,48 +58,59 @@ def _real_extract(self, url):
-            self.raise_geo_restricted()
+            restriction = settings.get('restrictionReason')
+            if restriction == 'RegionRestricted':
+                self.raise_geo_restricted()
 
-        if restriction and restriction != 'None':
-            raise ExtractorError(
-                '%s said: %s' % (self.IE_NAME, restriction), expected=True)
+            if restriction and restriction != 'None':
+                raise ExtractorError(
+                    '%s said: %s' % (self.IE_NAME, restriction), expected=True)
 
-        formats = []
-        subtitles = {}
-        for rendition_id, rendition in settings['renditions'].items():
-            bitrates = rendition.get('bitrates')
-            if not isinstance(bitrates, dict):
-                continue
-            m3u8_url = url_or_none(bitrates.get('hls'))
-            if not m3u8_url:
-                continue
-            formats.extend(self._extract_m3u8_formats(
-                m3u8_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                m3u8_id='%s-hls' % rendition_id, fatal=False))
-            cc_files = rendition.get('ccFiles')
-            if not isinstance(cc_files, list):
-                continue
-            for cc_file in cc_files:
-                if not isinstance(cc_file, list) or len(cc_file) < 3:
+            for rendition_id, rendition in settings['renditions'].items():
+                bitrates = rendition.get('bitrates')
+                if not isinstance(bitrates, dict):
+                    continue
+                m3u8_url = url_or_none(bitrates.get('hls'))
+                if not m3u8_url:
                     continue
-                cc_lang = cc_file[0]
-                cc_url = url_or_none(cc_file[2])
-                if not isinstance(cc_lang, compat_str) or not cc_url:
+                formats.extend(self._extract_m3u8_formats(
+                    m3u8_url, video_id, 'mp4', entry_protocol='m3u8_native',
+                    m3u8_id='%s-%s-%s-%s-hls' % (version, audio, extra, caption), fatal=False))
+                cc_files = rendition.get('ccFiles')
+                if not isinstance(cc_files, list):
                     continue
-                subtitles.setdefault(cc_lang, []).append({
-                    'url': cc_url,
-                })
+                for cc_file in cc_files:
+                    if not isinstance(cc_file, list) or len(cc_file) < 3:
+                        continue
+                    # Use this in future when the subtitle code can handle taking the name into account.
+                    # cc_lang = cc_file[0]
+                    cc_name = cc_file[1].replace(' ', '-').lower()
+                    cc_url = url_or_none(cc_file[2])
+                    if not isinstance(cc_name, compat_str) or not cc_url:
+                        continue
+                    subtitles.setdefault(cc_name, []).append({
+                        'url': cc_url,
+                    })
```

Comment:
```suggestion
                for cc_file in cc_files:
                    cc_url = url_or_none(try_get(cc_file, lambda x: x[2]))
                    # name is used since we cant distinguish subs with same language code
                    cc_lang = try_get(cc_file, (lambda x: x[1], lambda x: x[0]), str)
                    if cc_url and cc_lang:
                        subtitles.setdefault(cc_lang, []).append({'url': cc_url})
```

Reply:
This wouldn't convert cc_lang to lower and replace space with `-`, right?

Reply:
my bad. edited

Reply:
I also removed the cc_files list check